plugins/node/opentelemetry-instrumentation-redis-4/src/instrumentation.ts

```diff
@@ -0,0 +1,408 @@
+            const currCommandRes = redisRes[i];
+            if (currCommandRes instanceof Error) {
+              plugin._endSpanWithResponse(
+                span,
+                commandName,
+                commandArgs,
+                null,
+                currCommandRes
+              );
+            } else {
+              plugin._endSpanWithResponse(
+                span,
+                commandName,
+                commandArgs,
+                currCommandRes,
+                undefined
+              );
+            }
+          }
+        });
+        return execRes;
+      };
+    };
+  }
+
+  private _getPatchMultiCommandsAddCommand() {
+    const plugin = this;
+    return function addCommandWrapper(original: Function) {
+      return function addCommandPatch(this: any, args: Array<string | Buffer>) {
+        return plugin._traceClientCommand(original, this, arguments, args);
+      };
+    };
+  }
+
+  private _getPatchRedisClientMulti() {
+    return function multiPatchWrapper(original: Function) {
+      return function multiPatch(this: any) {
+        const multiRes = original.apply(this, arguments);
+        multiRes[MULTI_COMMAND_OPTIONS] = this.options;
+        return multiRes;
+      };
+    };
+  }
+
+  private _getPatchRedisClientSendCommand() {
+    const plugin = this;
+    return function sendCommandWrapper(original: Function) {
+      return function sendCommandPatch(
+        this: any,
+        args: Array<string | Buffer>
+      ) {
+        return plugin._traceClientCommand(original, this, arguments, args);
+      };
+    };
+  }
+
+  private _traceClientCommand(
+    origFunction: Function,
+    origThis: any,
+    origArguments: IArguments,
+    redisCommandArguments: Array<string | Buffer>
+  ) {
+    const hasNoParentSpan = trace.getSpan(context.active()) === undefined;
+    if (hasNoParentSpan && this._config?.requireParentSpan) {
+      return origFunction.apply(origThis, origArguments);
+    }
+
+    const clientOptions = origThis.options || origThis[MULTI_COMMAND_OPTIONS];
+
+    const commandName = redisCommandArguments[0] as string; // types also allows it to be a Buffer, but in practice it only string
+    const commandArgs = redisCommandArguments.slice(1);
+
+    const dbStatementSerializer =
+      this._config?.dbStatementSerializer || defaultDbStatementSerializer;
+
+    const attributes = {
+      [SemanticAttributes.DB_SYSTEM]: DbSystemValues.REDIS,
+      [SemanticAttributes.NET_PEER_NAME]: clientOptions?.socket?.host,
+      [SemanticAttributes.NET_PEER_PORT]: clientOptions?.socket?.port,
+      [SemanticAttributes.DB_CONNECTION_STRING]: clientOptions?.url,
+    };
+
+    try {
+      const dbStatement = dbStatementSerializer(commandName, commandArgs);
+      if (dbStatement != null) {
+        attributes[SemanticAttributes.DB_STATEMENT] = dbStatement;
+      }
+    } catch (e) {
+      this._diag.error('dbStatementSerializer throw an exception', e);
```

Comment:
It might make it easier to debug if we add the commandName and commandArgs

Reply:
added `commandName`. Args can contain non-bound payload size, and sensitive information, so I didn't add it here, but I amo open to discussing it if you think otherwise.